### PR TITLE
general accessibility fixes

### PR DESCRIPTION
**Pull request by floridemai**

### CODE REVIEW

**Review by JohnAGill — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ module.exports = {
   parser: '@typescript-eslint/parser',
   plugins: ['@typescript-eslint'],
   rules: {
-    'comma-dangle': 'off'
+    'comma-dangle': 'off',
+    'react-native-a11y/has-accessibility-hint': 'off'
```
Comment (ShaunBaker) [inline .eslintrc.js:8]: Why this change?

File: src/components/atoms/button.tsx [hunk 2]
```diff
@@ -85,7 +85,7 @@ export const Button: React.FC<ButtonProps> = ({
           type === 'empty' && styles.buttonEmpty
         ]}
         accessibilityRole="button"
-        importantForAccessibility="yes"
+        accessibilityState={{disabled}}
```
Comment (ShaunBaker) [inline src/components/atoms/button.tsx:88]: nit: Are the double curlies required?

File: src/components/atoms/card.tsx [hunk 3]
```diff
@@ -15,62 +20,71 @@ interface CardProps {
...
+              }
+            />
+          </View>
+        )}
+      </View>
+    );
+    return onPress ? (
+      <TouchableWithoutFeedback
+        ref={ref}
+        accessibilityRole="button"
```
Comment (ShaunBaker) [inline src/components/atoms/card.tsx:79]: No label?

File: .eslintrc.js (hunk 1 above)
Comment (floridemai) [reply .eslintrc.js:8]: It forces you to add both label and hint not just one which is not the case in quite a few places

File: src/components/atoms/button.tsx (hunk 2 above)
Comment (floridemai) [reply src/components/atoms/button.tsx:88]: Yes.
```
export interface AccessibilityState {
    /**
     * When true, informs accessible tools if the element is disabled
     */
    disabled?: boolean;
    /**
     * When true, informs accessible tools if the element is selected
     */
    selected?: boolean;
    /**
     * For items like Checkboxes and Toggle switches, reports their state to accessible tools
     */
    checked?: boolean | 'mixed';
    /**
     *  When present, informs accessible tools if the element is busy
     */
    busy?: boolean;
    /**
     *  When present, informs accessible tools the element is expanded or collapsed
     */
    expanded?: boolean;
}
```

File: src/components/atoms/card.tsx (hunk 3 above)
Comment (floridemai) [reply src/components/atoms/card.tsx:79]: It would be ideal to have the label in there but I don't know which one is it. We might want to pass in label (and hint) if reading the cards is not ideal.

File: .eslintrc.js (hunk 1 above)
Comment (ShaunBaker) [reply .eslintrc.js:8]: But this is a recommended best practice - hence the question.

I dont believe we have disabled this on other apps

Comment (ShaunBaker) [reply .eslintrc.js:8]: https://github.com/FormidableLabs/eslint-plugin-react-native-a11y/blob/master/docs/rules/has-accessibility-hint.md#has-accessibility-hint

Comment (ShaunBaker) [reply .eslintrc.js:8]: I'd encourage maintaining use of this rule
